homeassistant/components/home_connect/__init__.py

```diff
@@ -158,16 +164,55 @@ async def _async_service_program(call, method):
                 option[ATTR_UNIT] = option_unit
 
             options.append(option)
-
-        appliance = _get_appliance(hass, device_id)
-        await hass.async_add_executor_job(getattr(appliance, method), program, options)
+        try:
+            appliance = _get_appliance(hass, device_id)
+        except (ValueError, AssertionError) as err:
```

Comment:
This looks repetitive, with the same exception handled in the same way in multiple services

I would also normally suggest to move all the services to a separate module `services.py`:
```python
from .services import register_services

async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
    """Set up Home Connect component."""
    register_services(hass)

    return True
```

Reply:
Well, if `_get_appliance` method was only called from actions, we could add the exception handing to the function, but because is also called at `async_get_device_diagnostics`, I think it would not make too much sense to raise a validation error there.

Reply:
Btw, I agree with you on moving the services to `services.py`, I will do a PR with that, but I think this PR has higher priority

Reply:
> That's why I suggest adding as a new helper `_get_known_appliance` (or `_get_service_appliance`) dedicated to services

Ops, I didn't refreshed the page before sending to see your edited message